dataset/src/main/scala/frameless/ops/ColumnTypes.scala

```diff
@@ -3,17 +3,25 @@ package ops
 
 import shapeless._
 
-trait ColumnTypes[V, U <: HList] {
+/** A type class to extract the column types out of an HList of [[frameless.TypedColumn]].
+  *
+  * @example
+  * {{{
+  *   type U = TypedColumn[T,A] :: TypedColumn[T,B] :: TypedColumn[T,C] :: HNil
+  *   type Out = A :: B :: C :: HNil
+  * }}}
+  */
+trait ColumnTypes[T, U <: HList] {
   type Out <: HList
 }
 
 object ColumnTypes {
-  type Aux[V, U <: HList, Out0 <: HList] = ColumnTypes[V, U] {type Out = Out0}
+  type Aux[T, U <: HList, Out0 <: HList] = ColumnTypes[T, U] {type Out = Out0}
 
   implicit def deriveHNil[T]: ColumnTypes.Aux[T, HNil, HNil] = new ColumnTypes[T, HNil] { type Out = HNil }
 
-  implicit def deriveCons[V, H, TT <: HList, T <: HList](
-    implicit tail: ColumnTypes.Aux[V, TT, T]
-  ): ColumnTypes.Aux[V, TypedColumn[V, H] :: TT, H :: T] =
-    new ColumnTypes[V, TypedColumn[V, H] :: TT] {type Out = H :: T}
+  implicit def deriveCons[T, Head, Tail <: HList, ColTypes <: HList](
```

Comment:
Hey @kanterov, you mean about the head and tail? Yeah, that one I guess makes sense to use the standard H and TT. Overall the main goal of the  rename was to make it a bit easier to understand the code. Maybe we can revert the Head/Tail back to their standard notations and leave the ColTypes as is (since it has a well defined semantic in our case)?